swarm/swap_test.go

```diff
@@ -0,0 +1,406 @@
+		"swarm": func(ctx *adapters.ServiceContext, bucket *sync.Map) (s node.Service, cleanup func(), err error) {
+			config := api.NewConfig()
+			config.Port = strconv.Itoa(8500 + rand.Intn(9999))
+
+			dir, err := ioutil.TempDir("", "swap-network-test-node")
+			if err != nil {
+				return nil, nil, err
+			}
+			cleanup = func() {
+				err := os.RemoveAll(dir)
+				if err != nil {
+					log.Error("cleaning up swarm temp dir", "err", err)
+				}
+			}
+
+			config.Path = dir
+
+			privkey, err := crypto.GenerateKey()
+			if err != nil {
+				return nil, cleanup, err
+			}
+
+			config.Init(privkey)
+
+			//set Swap to be enabled for this test
+			config.SwapEnabled = true
+
+			swarm, err := NewSwarm(config, nil)
+			if err != nil {
+				return nil, cleanup, err
+			}
+
+			bucket.Store(bucketKeySwarm, swarm)
+			log.Info("new swarm", "bzzKey", config.BzzKey, "baseAddr", fmt.Sprintf("%x", swarm.bzz.BaseAddr()))
+			return swarm, cleanup, nil
+		},
+	})
+	defer sim.Close()
+
+	ctx := context.Background()
+	files := make([]file, 0)
+
+	var checkStatusM sync.Map
+	var nodeStatusM sync.Map
+	var totalFoundCount uint64
+
+	//connect all nodes in a chain
+	_, err := sim.AddNodesAndConnectChain(nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	//run the simulation
+	result := sim.Run(ctx, func(ctx context.Context, sim *simulation.Simulation) error {
+		//wait for kademlia to be healthy
+		if _, err := sim.WaitTillHealthy(ctx, 2); err != nil {
+			return err
+		}
+
+		nodeIDs := sim.UpNodeIDs()
+		shuffle(len(nodeIDs), func(i, j int) {
+			nodeIDs[i], nodeIDs[j] = nodeIDs[j], nodeIDs[i]
+		})
+		//upload a file for every node
+		for _, id := range nodeIDs {
+			item, ok := sim.NodeItem(id, bucketKeySwarm)
+			if !ok {
+				log.Error("No swarm")
+				return errors.New("No swarm")
+			}
+			swarm := item.(*Swarm)
+			key, data, err := uploadFile(swarm)
+			if err != nil {
+				return err
+			}
+			log.Trace("file uploaded", "node", id, "key", key.String())
+			files = append(files, file{
+				addr:   key,
+				data:   data,
+				nodeID: id,
+			})
+		}
+
+		// File retrieval check is repeated until all uploaded files are retrieved from all nodes
+		// or until the timeout is reached.
+		for {
+			if retrieve(sim, files, &checkStatusM, &nodeStatusM, &totalFoundCount) == 0 {
+				break
+			}
+		}
+		//every node has a map to all nodes it had interactions
+		//each entry in the map is a map of the other node with all the balances
+		balancesMap := make(map[enode.ID]map[enode.ID]int64)
+
+		//iterate all nodes
+		for _, node := range sim.NodeIDs() {
+			item, ok := sim.NodeItem(node, bucketKeySwarm)
+			if !ok {
+				log.Error("No swarm")
+				return errors.New("No swarm")
+			}
+			swarm := item.(*Swarm)
+
+			//submap for each node is a map of all nodes with the balance for that node
+			subBalances := make(map[enode.ID]int64)
+
+			//iterate all nodes again...
+			//get all balances with other peers for every node
+			for _, n := range sim.NodeIDs() {
```

Comment:
why do you need maps, why dont we just iterate 


```golang

for _, x := range sim.NodeIDs() {
  	xs, ok := sim.NodeItem(x, bucketKeySwarm)
	for _, y := range sim.NodeIDs() {
		ys, ok := sim.NodeItem(y, bucketKeySwarm)
                xbaly := xs.swap.GetPeerBalance(y)                
                ybalx := ys.swap.GetPeerBalance(x)
                // compare 

```

Reply:
Because then I can use it for nice printing if `*printStats == true`